features/FxA-84-recaptcha/README.md

```diff
@@ -3,35 +3,163 @@
+  to implement our own.
+
+* Some of our anti-abuse measures are not based on the number of
+  requests, e.g. the planned use of an IP blocklist.  In these cases
+  the existing "attempt limit exceeded" message would be inappropriate
+  and confusing.
 
 ****
 
 ## Detailed design
 
-### ReCAPTCHA View
-<img src='example_expired.png' alt='recaptcha' width='400' />
+### UX
+
+We currently have a single, unfriendly, dead-end experience
+for users that trigger our fraud-and-abuse system:
+
+<img src='attempt_limit_exceeded.png' width='400' />
+
+Instead, we should distinguish three separate cases
+and provide clear messaging for each:
+
+1. This particular user has submitted too many requests
+   too quickly, and they must wait before trying again.
+2. Other users on this IP address have submitted too many
+   requests recently, so they must wait before trying again.
+3. The request was blocked because it looked likely to be from a bot,
+   but would be accepted if the user can demonstrate that they aren't.
+
+For case (1) we should present a more friendly error message
+along with a countdown telling them how long to wait:
+
+<img src='wait_this_user.png' width='400' />
+
+For case (2) we should explain the situation
+in a way that doesn't imply it is the user's fault,
+and show a countdown telling them how long to wait:
+
+<img src='wait_this_ip.png' width='400' />
+
+For case (3) we should explain the situation
+and offer the user the option of completing a captcha
+to continue with the request:
+
+<img src='offer_captcha.png' alt='recaptcha' width='400' />
+
+If the user chooses to do so,
+we load recaptcha into the page like so:
+
+<img src='show_captcha.png' alt='recaptcha' width='400' />
+
+And the user must complete the captcha
+before the sign-in button is re-enabled.
+
+Notes:
+
+* This aren't intended to be the final screens or strings.
+  Don't let rfkelly design screens or write strings.
+* The time to wait is returned in the error response from the server.
+* Distinguishing case (3) will require the server to return a new error
+  code rather than the existing errno 114 "attempt limit exceeded"; we can
+  use errno 125 "request blocked" for this purpose.
+* Distinguishing case (2) from case (1) will require the content-server
+  to keep track of whether the user has made any attempts in this session,
+  since the server doesn't have enough info to tell the difference.
+* The treatment here is for the sign-in form,
+  but there's no reason we couldn't take a similar approach
+  on other views subject to customs checking,
+  such as account creation or password reset.
+* The idea behing a multi-click process for the captcha, is to ensure the
+  user takes some deliberate action before we add a third-party javascript
+  file into the page.  This may help reduce risk in shipping the initial
+  version and we can revisit it in a followup if the feature is successful.
+  But I know it's likely to be worse UX, so it's just a thought...
```

Comment:
In fact, I have some recollection of a previous version of the captcha prototype that did this, so shoot it down if I'm retreading ground that's already well-trodden...